packages/core/src/browser/keybinding.ts

```diff
@@ -279,23 +339,43 @@ export class KeybindingRegistry {
         return result;
     }
 
+    /**
+     * Returns wether a binding is being shadowed or not inside scope
+     *
+     * @param binding binding to look if shadowed
+     * @param scope scope where to look
```

Comment:
I think the doc is outdated, as it doesn't take the scope into account anymore with your changes.